File: app/src/main/java/com/innoveworkshop/gametest/assets/DroppingRectangle.java
package com.innoveworkshop.gametest.assets;

import com.innoveworkshop.gametest.engine.Rectangle;
import com.innoveworkshop.gametest.engine.Vector;

public class DroppingRectangle extends Rectangle {
    float dropRate = 0;

    public DroppingRectangle(Vector position, float width, float height, float dropRate, int color) {
        super(position, width, height, color);
        this.dropRate = dropRate;
    }

    @Override
    public void onFixedUpdate() {
        super.onFixedUpdate();

        if (!isFloored())
            position.y += dropRate;
    }
}

File: app/src/main/java/com/innoveworkshop/gametest/assets/DroppingRectangle.kt
package com.innoveworkshop.gametest.assets

import com.innoveworkshop.gametest.engine.Rectangle
import com.innoveworkshop.gametest.engine.Vector

class DroppingRectangle(
    position: Vector?,
    width: Float,
    height: Float,
    dropRate: Float,
    color: Int
) : Rectangle(position, width, height, color) {
    var dropRate: Float = 0f

    init {
        this.dropRate = dropRate
    }

    override fun onFixedUpdate() {
        super.onFixedUpdate()

        if (!isFloored) position.y += dropRate
    }
}
